### ISSUE TITLE
#Feature Request, display Properties of Relationship between nodes.

### ISSUE THREAD

**Issue by vence-andersen** [neo4j/neo4j-python-driver#722]
After running a code with session.run(query), if we pass .data() to the output, we get the Properties of the output nodes but not the Properties of the relationship between the output nodes. Adding that will be helpful.

**Comment by robsdedude** [comment]
To give some more context:

```python
import neo4j


def work(tx, query, **params):
    result = tx.run(query, params)
    return result.data()


def main():
    auth = ("neo4j", "pass")
    uri = "neo4j://localhost:7687"
    with neo4j.GraphDatabase.driver(uri, auth=auth) as driver:
        with driver.session() as session:
            data = session.write_transaction(
                work,
                "MERGE (n:Source {name: 'foo'})"
                "-[r:REL {name: 'bar'}]->"
                "(m:Sink {name: 'baz'}) "
                "RETURN n, r, m"
            )
    print(data)


if __name__ == "__main__":
    main()
```

Gives the following output:
```python
[{'n': {'name': 'foo'}, 'r': ({'name': 'foo'}, 'REL', {'name': 'baz'}), 'm': {'name': 'baz'}}]
```

Not only, does this not contain properties for relationships, it also doesn't contain the labels of the nodes.

Generally, I recommend to not use `.data` for anything more than tinkering around. It's a quick prototyping API, really. If we were to change the return value of `Result.data` this would most likely break projects that rely on the format that it returns currently.

If you look into the implementation of `Result.data`, you can see that it's actually not very complicated. You could also write your own custom data serialization that does exactly what you need.

`Result.data`:

https://github.com/neo4j/neo4j-python-driver/blob/c79d62eb0e6c8b49391b96435616f58b2c9e0824/neo4j/work/result.py#L365-L374

which calls `Record.data`

https://github.com/neo4j/neo4j-python-driver/blob/1ed96f94a7a59f49c473dadbb81715dc9651db98/neo4j/data.py#L239-L251

which uses

https://github.com/neo4j/neo4j-python-driver/blob/c79d62eb0e6c8b49391b96435616f58b2c9e0824/neo4j/data.py#L236-L265

TL;DR: won't fix but relatively easy to implement yourself.

If you have further questions, don't hesitate to ask.
